EIPS/eip-2294.md

```diff
@@ -0,0 +1,40 @@
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-09-19
+requires: 155
+---
+
+## Abstract
+This EIP restricts the size of [EIP-155](./eip-155.md) Chain ID parameter to a particular maximum value `floor(MAX_UINT64 / 2) - 36`, in order to ensure that there is a standard around how this parameter is to be used between different projects.
+
+## Motivation
+EIP-155 introduces the Chain ID parameter, which is an important parameter used for domain separation (replay protection) of Ethereum protocol signed messages. However, it does not specify any properties about the size that this parameter takes. @axic mentions this concern in [EIP-1344 (comment)](https://ethereum-magicians.org/t/eip-1344-add-chain-id-opcode/1131/109) and suggests a reasonable maximum enforced size in order to ensure that there are no issues when encoding this parameter. This would allow a sufficient amount of different values for this parameter, which is typically chosen by community consensus as a genesis parameter for a given chain and thus does not change often.
+
+There have been suggestions of using a hash-based identifier in place on Chain ID to allow the value to adapt over time to different contentious forks and other scenarios. This proposal does not describe this behavior, but ~63 bits of entropy should be enough to ensure that no collisions are likely for reasonable (e.g. non-malicious) uses of this feature for that purpose.
+
+## Specification
+The maximum value of Chain ID is `9,223,372,036,854,775,771` (`MAX_CHAIN_ID`). This value is `floor(MAX_UINT64 / 2) - 36`, and is chosen to avoid overflow when performing uint64 math. For reference, a value of 0 or less is also disallowed.
+
+A client should reject a value outside of this range in a provided transaction, and disallow a genesis configuration with a value for Chain ID outside of this limit. Due to how the calculation for chain ID is performed, the maximum value seen during the arithmetic is `CHAIN_ID * 2 + 36`, so clients must test to ensure no overflow conditions are encountered when the highest value is used. No underflow is possible.
+
+## Rationale
+Without a well-chosen value of Chain ID, there could be differences in the implementation of EIP-155 (and EIP-1344 by derivative) in both client codebase and external tooling that could lead to consensus-critical vulnerabilities being introduced to the network. By making this limit explicit, we avoid this scenario for Ethereum and any project which uses the Ethereum codebase.
+
+## Backwards Compatibility
+This EIP introduces a change that affects previous implementations of this feature. However, since no known chain makes use of a value outside of the suggested bounds, there should not be an issue in adopting this limit on the size of this parameter, therefore the impact should be non-existent.
+
+## Test Cases
+[Aleth PR](https://github.com/ethereum/aleth/pull/5742)
```

Comment:
Requesting a manual approval from editors for including this link
```
https://github.com/ethereum/aleth/pull/5742
```
@axic @SamWilsn @lightclient @gcolvin @Pandapip1

Reply:
I'd prefer if we include test cases in the `assets` directory.

Reply:
it is an Ethereum project

Reply:
No, we have agreed that this is not allowed.

Reply:
Fine. I will remove the links for now until we have a clearer external link policy.

For the record, I respectively protest your interpretation of our link policy, my friend @Pandapip1 .